# Ejercicio 11


Encuentre la solución general del sistema de ecuaciones:

$$ x' = \begin{pmatrix} 2 & 2 + i \\ -1 & -1 - i \end{pmatrix}x   $$


## Solución analítica 

$$ A = \begin{pmatrix} 2 & 2 + i \\ -1 & -1 - i \end{pmatrix} \quad \Rightarrow \quad A-rI = \begin{pmatrix} 2 -r & 2 + i \\ -1 & -1 - i - r \end{pmatrix} $$

$$ det(A-rI) = 0 \quad \quad \iff \quad \quad  (2-r)(-1-i-r) + (2+i) = 0 $$

$$ -2-2i-2r+r+ri+r^{2} + 2 + i = 0 $$

$$ r^{2} + (i-1)r - i = 0 $$

$$ (r+i)(r-1) = 0 $$

$$ r_{1} = -i \quad \quad \vee \quad \quad  r_{2} = 1 $$

### Determinación de $\xi^{(1)}$ 

$$ (A - r_{1}I)\xi^{(1)} = 0 \quad \iff \quad  \begin{pmatrix} 2 - r_{1} & 2 + i \\ -1 & -1 - i - r_{1} \end{pmatrix}\xi^{(1)} = \begin{pmatrix} 0 \\ 0 \end{pmatrix} $$

$$  \begin{pmatrix} 2 + i & 2 + i \\ -1 & -1 - i + i\end{pmatrix}\xi^{(1)} = \begin{pmatrix} 0 \\ 0 \end{pmatrix}  $$

$$  \begin{pmatrix} 2 + i & 2 + i \\ -1 & -1 \end{pmatrix}\xi^{(1)} = \begin{pmatrix} 0 \\ 0 \end{pmatrix}  $$

$$-\xi^{(1)}_{1} - \xi^{(1)}_{2} = 0 $$

$$ \xi^{(1)}_{2} = - \xi^{(1)}_{2}  $$

Entonces 

$$\xi^{(1)} = \begin{pmatrix} 1 \\ -1 \end{pmatrix} $$


Normalicemos a $\xi^{(1)}$:

$$ u_{\xi1} = \frac{1}{\sqrt{2}}\begin{pmatrix} 1 \\ -1 \end{pmatrix}  $$


In [14]:
import numpy as np 
A = np.array([[2,2+1j],[-1,-1-1j]])
r,xi = np.linalg.eig(A) 

In [15]:
r[1]

(-9.633732540219107e-17-0.9999999999999998j)

In [18]:
np.linalg.norm(xi[:,0]), xi[:,1]

(1.0000000000000002,
 array([ 0.70710678+0.00000000e+00j, -0.70710678+1.87141271e-16j]))

De donde los eigenvalores de la matriz $A$ son $r_{1} = 1$ y $r_{2} = -i$

y también 

$$\xi^{(1)} = \begin{pmatrix}0.91 \\ -0.37 + 0.18i \end{pmatrix}, \quad \quad \xi^{(2)} = \begin{pmatrix} 0.707 \\ -0.707 \end{pmatrix} $$


In [19]:
def solu_gen(t,c1,c2):
    x = c1*xi[:,0]*np.exp(r[0]*t) + c2*xi[:,1]*np.exp(r[1]*t) 
    return x

In [20]:
solu_gen(0,1,1) 

array([ 1.61997771+0.j        , -1.07225515+0.18257419j])